### DISCUSSION TITLE
Using existing clustering reference

### DISCUSSION THREAD

**Discussion by vcglinnaeus** [satijalab/seurat#4233] [Q&A] [answered]
Is it possible to assign a cluster instead of using the clustering algorithm in the package? I love Seurat and it is easy to use but for cases where biologists have prepared a clustered data as a reference, how can I use them. I tried reassigning it to active.idents, but I guess it doesnt work like that?

**Comment by f6v** [comment, marked as answer]
You could assign clusters/cell types by doing so:
```
your_cell_types <- ... # load from file
your_seurat_obj$cell_type <- your_cell_types
```
And then you can subset Seurat object:
```
your_seurat_obj[, your_seurat_obj$celltype == "Neuron"]
```

**Reply by vcglinnaeus** [reply]
This one has no association with clusters yet. I am trying to do a pca, via elbow joint i can see that that optimal k is 15. but i need the clusters to assume a k equivalent to the given clustering (data from wet lab). Thank you. And also I love Seurat it helped me a lot learning sc

**Reply by f6v** [reply]
If you specifically want to assign clusters, couldn't you use `Idents`?
```
Idents(your_seurat_obj) <- ... # vector of identities
```

**Reply by timoast** [reply]
@vcglinnaeus the elbow plot is used for determining the number of PCs to include in downstream analysis steps, not the number of clusters present in the data. I'm not completely sure what you're asking, but you might find the reference-based mapping vignette useful: https://satijalab.org/seurat/articles/integration_mapping.html

**Comment by vcglinnaeus** [comment]
Good day,

I have tried it. But the number of clusters from the seurat clustering
doesnt match mine. Well I tried specifying k = 33 (for example) but im not
getting a cluster of k=33 but of 10,11 or 12 instead. It is like following
the optimal k as observed from the elbow joint.

**Reply by timoast** [reply]
Seurat does not use k-means clustering, but uses a graph-based community detection method. The `k.param` parameter in `FindNeighbors` refers to the number of nearest neighbors to find. Please see the documentation for FindNeighbors and FindClusters for more information.
